### ISSUE TITLE
Outline is broken for symbols defined in macros

### ISSUE THREAD

**Issue by hokein** [clangd/clangd#475]
clangd-version: head c0bd9fa137c28a3ef833b46b7f9770b060275281

1. Open `clang/include/clang/AST/RecursiveASTVisitor.h`
2. Open the outline panel in VSCode
3. click the `TraverseAddrLabelExpr` symbol.

actual result: jump to Line 617 which doesn't makes many sense.

expected result: should jump to L2557 (`DEF_TRAVERSE_STMT(AddrLabelExpr, {})`). 

the issue seems to be related to macros.

**Comment by HighCommander4** [comment]
>    3. click the `TraverseAddrLabelExpr` symbol.

I don't have a symbol named just `TraverseAddrLabelExpr`, only `RecursiveASTVisitor<Derived>::TraverseAddrLabelExpr`. Clicking on that gives me the "expected result".

**Comment by hokein** [comment]
> I don't have a symbol named just TraverseAddrLabelExpr, only RecursiveASTVisitor<Derived>::TraverseAddrLabelExpr.

yeah, I missed the qualifiers. I meat `RecursiveASTVisitor<Derived>::TraverseAddrLabelExpr.`

> Clicking on that gives me the "expected result".

hmm, interesting, which version are you using? We may have a regression.

**Comment by HighCommander4** [comment]
> hmm, interesting, which version are you using?

I was running revision a378c0449507e00e96534ff9ce9034e185425182 (trunk from ~ Jul 5) when testing this.

**Comment by HighCommander4** [comment]
`TraverseAddrLabelExpr` works for me with a very recent trunk as well. However, if I try some other functions, the results are hit and miss: sometimes accurate, and sometimes bogus. Very strange.

**Comment by hokein** [comment]
yeah, this is wired. I also notice that it happens on gtests macros if you try it on a unittest file.

**Comment by HighCommander4** [comment]
This does not seem to be fixed by #500. There's probably something else going on here.
